app/gui2/src/components/NavBreadcrumb.vue

```diff
@@ -1,10 +1,27 @@
 <script setup lang="ts">
-const props = defineProps<{ text: string; active: boolean }>()
+import { ref, watch, type ComponentInstance } from 'vue'
+import AutoSizedInput from './widgets/AutoSizedInput.vue'
+
+const model = defineModel<string>({ required: true })
+const _props = defineProps<{ active: boolean; editing: boolean }>()
+const emit = defineEmits<{ renamed: [newName: string] }>()
+
+const input = ref<ComponentInstance<typeof AutoSizedInput>>()
+watch(input, (input, old) => {
+  if (old == null && input != null) input.focus()
+})
 </script>
 
 <template>
-  <div :class="['NavBreadcrumb', { inactive: !props.active }]">
-    <span v-text="props.text"></span>
+  <div :class="['NavBreadcrumb', { inactive: !active }]">
+    <AutoSizedInput
+      v-if="editing"
+      ref="input"
+      v-model.lazy="model"
+      :autoSelect="true"
+      @change="emit('renamed', $event ?? '')"
```

Comment:
When the AutoSizedInput changes, this component emits both `update:modelValue` and `renamed`?

Reply:
Yes, the renamed event is redundant, removed.